### PR TITLE
ADEN-10998 | ucp-mobile: UAP 3.0 BFAA - stickyBig to stickySmall on scroll

### PR DESCRIPTION
**Pull request by Loczi94**

### CODE REVIEW

File: platforms/shared/index.ts [hunk 1]
```diff
@@ -62,22 +63,24 @@ export * from './templates/handlers/slot/slot-position-sticky-small-handler';
 export * from './templates/handlers/slot/slot-size-impact-handler';
 export * from './templates/handlers/slot/slot-size-impact-to-resolved-handler';
 export * from './templates/handlers/slot/slot-size-resolved-handler';
+export * from './templates/handlers/slot/slot-size-stickyBig-to-stickySmall-handler';
```
Comment (nandy-andy) [inline platforms/shared/index.ts:66]: Is the plan to replace `'./templates/handlers/slot/slot-decision-sticky-big-to-sticky-small-handler'` with `'./templates/handlers/slot/slot-size-stickyBig-to-stickySmall-handler'` in the future? 🤔  The names are so similar that it's confusing ;)

File: platforms/shared/templates/handlers/slot/slot-decision-sticky-big-to-sticky-small-handler.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	TemplateTransition,
+} from '@wikia/ad-engine';
+import { Inject, Injectable } from '@wikia/dependency-injection';
+import { Subject } from 'rxjs';
+import { filter, startWith, takeUntil, tap } from 'rxjs/operators';
+import { ScrollCorrector } from '../../helpers/scroll-corrector';
+import { UapDomReader } from '../../helpers/uap-dom-reader';
+
+@Injectable({ autobind: false })
+export class SlotDecisionStickyBigToStickySmallHandlerHandler implements TemplateStateHandler {
```
Comment (nandy-andy) [inline platforms/shared/templates/handlers/slot/slot-decision-sticky-big-to-sticky-small-handler.ts:15]: Do we need to keep `HandlerHandler` in the name?

Comment (Loczi94) [reply platforms/shared/templates/handlers/slot/slot-decision-sticky-big-to-sticky-small-handler.ts:15]: No, `HandlerHandler` is here by mistake.

File: platforms/shared/templates/handlers/slot/slot-size-sticky-big-to-sticky-small-handler.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	constructor(private domListener: DomListener, private manager: UapDomManager) {}
+
+	async onEnter(): Promise<void> {
+		const adSlotPlaceholderTopOffset = this.manager.getAdSlotPlaceholderTopOffset();
+
+		this.manager.setImpactImage();
+		merge(this.domListener.resize$)
+			.pipe(
+				startWith({}),
+				tap(() => {
```
Comment (preludek) [inline platforms/shared/templates/handlers/slot/slot-size-sticky-big-to-sticky-small-handler.ts:20]: isn't `tap(() => this.manager.setSlotHeightBigToSmall(adSlotPlaceholderTopOffset)),` the same thing?

File: platforms/shared/templates/helpers/uap-dom-reader.ts [hunk 4]
```diff
@@ -51,15 +55,52 @@ export class UapDomReader {
...
+
+		if (progress >= 1) {
+			return 1;
+		}
+
+		if (progress <= 0) {
+			return 0;
+		}
 
-		return progress >= 1 ? 1 : progress;
```
Comment (preludek) [inline platforms/shared/templates/helpers/uap-dom-reader.ts:62]: I don't mind that being a oneliner 🙈  I feel it's readable. 
`return progress >= 1 ? 1: 0;`

File: platforms/shared/templates/helpers/uap-dom-reader.ts [hunk 5]
```diff
@@ -51,15 +55,52 @@ export class UapDomReader {
...
+
+		if (progress >= 1) {
+			return 1;
+		}
+
+		if (progress <= 0) {
+			return 0;
+		}
+
+		return progress;
```
Comment (preludek) [inline platforms/shared/templates/helpers/uap-dom-reader.ts:82]: ditto :P

File: platforms/shared/templates/helpers/uap-dom-reader.ts [hunk 6]
```diff
@@ -25,6 +25,10 @@ export class UapDomReader {
 		);
 	}
 
+	getNavbarHeight(): number {
```
Comment (preludek) [inline platforms/shared/templates/helpers/uap-dom-reader.ts:28]: getNavbarOffsetHeight?

**Review by preludek — APPROVED** [review]

File: platforms/shared/templates/helpers/uap-dom-reader.ts (hunk 4 above)
Comment (jakubbanas) [reply platforms/shared/templates/helpers/uap-dom-reader.ts:62]: As we talked on Slack, I moved this logic to a separate function.

**Pull request closed without merging**